src/memory_provider.c

```diff
@@ -22,13 +22,13 @@
 #include "utils_assert.h"
 
```

Comment:
Where is support for defaults for memory providers? I see only implementation for pools, and we need both